File: src/main/golang/leetcode/editor/cn/[1]Two Sum.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func twoSum(nums []int, target int) []int {
	mapper := make(map[int]int)
	for i, num := range nums {
		if v, ok := mapper[target-num]; ok {
			return []int{v, i}
		}
		mapper[num] = i
	}
	return nil
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[1]Two Sum.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i, num in enumerate(nums):
            if target - num in dic:
                return [dic[target - num], i]
            else:
                dic[num] = i
# leetcode submit region end(Prohibit modification and deletion)
